src/learning/infer.py

```diff
@@ -0,0 +1,31 @@
+import torch
+import torchvision
+from torchvision.utils import save_image
+import utils
+from modules import Generator
+
+MODEL_PATH = 'gpu_model'
+VALIDATION_DATA_PATH = 'test_A'
+
+DISPLAY_IMAGE = False
+N_VALIDATION_IMAGES = 3
+N_RESIDUAL_BLOCKS = 9
+USE_DROPOUT = False
+IMAGE_SIZE = (256, 256)
+
+
+def main():
+    model = Generator(N_RESIDUAL_BLOCKS, USE_DROPOUT)
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    model.load_state_dict(torch.load(MODEL_PATH, map_location=device))
+    transform = utils.get_transform(IMAGE_SIZE)
+    validation_data = torchvision.datasets.ImageFolder(
+        root=VALIDATION_DATA_PATH, transform=transform)
+
+    for i in range(N_VALIDATION_IMAGES):
+        prediction = model(validation_data[i][0].unsqueeze(0))
+        save_image(prediction, 'prediction%d.png' % i, normalize=True)
```

Comment:
use string.format() instead of '%' as this is a deprecated string formatting method (https://docs.python.org/release/3.1.5/library/stdtypes.html#old-string-formatting-operations)
suggestion: 'prediction{}.png'.format(i)

Reply:
Good point! Since 3.6 f-strings are even more avant-garde and recommended.
Done.